docs/content/concepts/testing.mdx

```diff
@@ -135,22 +122,179 @@ def run_in_prod():
+    return config.my_string
+
+
+def test_asset_requires_config():
+    result = asset_requires_config(config=MyAssetConfig(my_string="foo"))
+    ...
+```
+
+### Testing assets with resources
+
+If your asset requires resources, you can specify them as arguments when invoking the asset directly.
+
+Consider the following asset, which requires a resource `bar`.
+
+```python file=/concepts/ops_jobs_graphs/unit_tests.py startafter=start_test_resource_asset endbefore=end_test_resource_asset
+from dagster import asset, ConfigurableResource
+
+
+class BarResource(ConfigurableResource):
+    my_string: str
+
+
+@asset
+def asset_requires_bar(bar: BarResource) -> str:
+    return bar.my_string
+
+
+def test_asset_requires_bar():
+    result = asset_requires_bar(bar=BarResource(my_string="bar"))
+    ...
+```
+
+#### Testing assets with complex resources
+
+In order to test assets which rely on complex resources, such as those that build separate clients, a common pattern is to use tools such as the `mock` library to fake your resource and associated client. See the section on [Testing ops with complex resources](#testing-ops-with-complex-resources) for an example.
+
+### Testing multiple assets together
+
+You may want to test multiple assets together, to more closely mirror actual materialization. This can be done using the <PyObject object="materialize_to_memory" /> method, which loads the materialized results of assets into memory:
+
+```python file=/concepts/ops_jobs_graphs/unit_tests.py startafter=start_materialize_asset endbefore=end_materialize_asset
+from dagster import asset, materialize_to_memory
+
+
+@asset
+def data_source():
+    return get_data_from_source()
+
+
+@asset
+def structured_data(data_source):
+    return extract_structured_data(data_source)
+
+
+# An example unit test using materialize_to_memory
+def test_data_assets():
+    result = materialize_to_memory([data_source, structured_data])
+    assert result.success
+    # Materialized objects can be accessed in terms of the underlying op
+    materialized_data = result.output_for_node("structured_data")
+    ...
+```
+
+Mock resources can be provided directly using `materialize_to_memory`:
+
+```python file=/concepts/ops_jobs_graphs/unit_tests.py startafter=start_materialize_resources endbefore=end_materialize_resources
+from dagster import asset, materialize_to_memory, ConfigurableResource
+import mock
+
+
+class MyServiceResource(ConfigurableResource): ...
+
+
+@asset
+def asset_requires_service(service: MyServiceResource): ...
+
+
+@asset
+def other_asset_requires_service(service: MyServiceResource): ...
+
+
+def test_assets_require_service():
+    # Mock objects can be provided directly.
+    result = materialize_to_memory(
+        [asset_requires_service, other_asset_requires_service],
+        resources={"service": mock.MagicMock()},
+    )
+    assert result.success
+    ...
+```
 
 ---
 
-## Examples
+## Testing asset checks
```

Comment:
This is the entirety of the testing section. I'm not sure how much we want to put in here, as other sections on this page are pretty light.

If you want to leave a few notes about anything that might be good, that'd be helpful.

Reply:
You could state "This is currently only supported with asset checks that don't use `context`, or use `OpExecutionContext`- this uses the `build_op_execution_context` fn in the `Testing ops  that rely on context` section below. We don't currently have a `build_asset_check_execution_context` fn.